generator/src/main/java/com/graphicsfuzz/generator/knownvaluegeneration/ExpressionGenerator.java

```diff
@@ -486,19 +488,55 @@ private Expr generateVariableFact(Value value,
                                     Stmt stmtToInsertBefore) {
     boolean atGlobalScope = factManager.globalScope() || generator.nextBoolean();
     final String varName = genVarName(value, atGlobalScope);
-
+    final Expr initializer = generateExpr(atGlobalScope ? globalFactManager : factManager,
+        currentFunction, stmtToInsertBefore, value);
+    // If generating global scope variables, initialisation would be performed later in a global
+    // initializer function.
     final VariableDeclInfo variableDeclInfo = new VariableDeclInfo(varName, null,
-        new Initializer(generateExpr(atGlobalScope ? globalFactManager : factManager,
-            currentFunction,
-            stmtToInsertBefore,
-            value
-        )));
-
+        atGlobalScope ? null : new Initializer(initializer));
     final VariablesDeclaration variablesDecl = new VariablesDeclaration(value.getType(),
         variableDeclInfo);
     final VariableDeclFact variableDeclFact = new VariableDeclFact(variablesDecl,
         variableDeclInfo, value);
+
     if (atGlobalScope) {
```

Comment:
-> "Search for an existing function for initialization of global variables.  If not found, we have to create one and add a call to the top of 'main' to invoke it."